low-level/solution/lr2/support_bundle/sspl_support_bundle

```diff
@@ -0,0 +1,388 @@
+
+        if file_path:
+            if os.path.exists(file_path):
+                file_path = file_path+"/sspl/"
+                self.tmp_bundle_path = os.path.normpath(file_path)
+            else:
+                msg = "Given path %s doesn't exist" % (file_path)
+                self.__clear_tmp_files()
+                raise SupportBundleError(1, msg)
+        os.makedirs(self.tmp_bundle_path, exist_ok=True)
+        try:
+            DEFAULT_STATE = Conf.get(
+                SSPL_CONF, f"{SYSTEM_INFORMATION}>{SSPL_STATE}", "active")
+            try:
+                sspl_state = open(
+                    f"/var/{sspl_const.PRODUCT_FAMILY}/sspl/data/state.txt")
+                sspl_state = sspl_state.readline().rstrip().split("=")
+                # Capturing enclosure logs only when sspl state is active
+                sspl_state = sspl_state[1] if len(
+                    sspl_state) == 2 else DEFAULT_STATE
+                if not sspl_state == 'active':
+                    logger.info("SSPL is in 'degraded' mode,"
+                                "so enclosure data not collected "
+                                "as part of this node support bundle.")
+
+            except (FileNotFoundError, OSError) as err:
+                logger.error("Failed to open the SSPL 'state.txt' "
+                             "file with an error '{0}',Can't determine "
+                             "the SSPL state, So enclosure logs also "
+                             "being collected.".format(err))
+                sspl_state = 'active'
+
+            self.get_ipmi_sel_data()
+            self.get_config_data()
+            self.get_drives_smart_data()
+            if sspl_state == "active" and not include_encl:
+                StorageEncl().get_enclosure_logs(self.enclosure_log, logger)
+
+            if os.path.exists(self.enclosure_log) and sspl_state == 'active' \
+                    and not include_encl:
+                enclosure_zip_file = "enclosure-wwn-{0}-logs-{1}.zip".format(
+                    self.enclosure_wwn, str(int(time.time())))
+                shutil.copy(self.enclosure_log,
+                            self.tmp_bundle_path+enclosure_zip_file)
+                logger.info("Enclosure Log File Location: %s" %
+                            self.tmp_bundle_path+enclosure_zip_file)
+
+        except OSError as err:
+            msg = "Facing problem while creating sspl "
+            "support bundle: {0}".format(err)
+            self.__clear_tmp_files()
+            raise SupportBundleError(1, msg)
+        try:
+            # Process request
+            BundleRequestHandler._update_request_status(
+                request_id, BundleRequestHandler.SUCCESS)
+        except Exception as err:
+            status = BundleRequestHandler.FAILED + f" - {err}"
+            BundleRequestHandler._update_request_status(
+                request_id, status)
+        return request_id
+
+    def get_ipmi_sel_data(self):
```

Comment:
ipmi_client can have this function

Reply:
This function is simple use of generic ipmi functions we have in ipmi_client. Not sure if we should move it in ipmi_client itself.

Reply:
SEL list is generic most usecase, where any ipmiclient can serve based on its command support. Its on similar lines of generic functions we have there already, like get_sensor_list_by_type() , get_manufacturer_name()

Reply:
It is not a generic function. here we are just dumping sel data into a file and adding this file into SB. So it is SB-specific code only.